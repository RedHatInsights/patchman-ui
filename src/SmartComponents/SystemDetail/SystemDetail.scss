.patchTabSelect {
    background-color: white;
    .pf-v5-c-tabs__item-text {
        padding: 8px 0px 8px 0px;
    }
}