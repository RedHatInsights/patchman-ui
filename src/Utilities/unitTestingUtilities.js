import { useSelector } from 'react-redux';

/* eslint-disable */
export const initMocks = () => {

    window.insights = {
        loadInventory: (args) => new Promise((resolve, reject) =>  {
            resolve(({
                inventoryConnector: (args) => {
                            const InventoryTable = ({ children }) => <div >A mock passed! {children} </div>;
                            return ({ InventoryTable });
                        },
                        mergeWithEntities: () => {}
            }));
        }),
        loadRemediations: () => new Promise((resolve, reject) =>  {
            resolve({ 
                openWizard: () => new Promise((resolve, reject) =>  { 
                    resolve(true)
                })
            });
        })
    };

};

export const mockStore = (initialState, mutatedState) => {
    const customMiddleWare = store => next => action => {
        useSelector.mockImplementation(callback => {
            return callback(mutatedState);
        });
        next(action);
    };

    const mockStore = configureStore([customMiddleWare]);

    return mockStore(initialState);
}
/* eslint-enable */
