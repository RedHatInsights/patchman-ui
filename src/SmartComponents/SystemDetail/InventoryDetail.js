import React, { useEffect } from 'react';
import { Main } from '@redhat-cloud-services/frontend-components/Main';
import { useSelector, useDispatch } from 'react-redux';
import Header from '../../PresentationalComponents/Header/Header';
import { paths } from '../../Routes';
import { register } from '../../store';
import { SystemDetailStore } from '../../store/Reducers/SystemDetailStore';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';
import { setPageTitle } from '../../Utilities/Hooks';
import { InventoryDetailHead, AppInfo, DetailWrapper } from '@redhat-cloud-services/frontend-components/Inventory';
import { Alert, Grid, GridItem, TextContent, Text } from '@patternfly/react-core';
import { fetchSystemDetailsAction } from '../../store/Actions/Actions';
import propTypes from 'prop-types';
import { clearNotifications } from '@redhat-cloud-services/frontend-components-notifications/redux';;
import PatchSetWizard from '../PatchSetWizard/PatchSetWizard';

const InventoryDetail = ({ match }) => {

    const [patchSetState, setBaselineState] = React.useState({
        isOpen: false,
        shouldRefresh: false,
        systemsIDs: []
    });

    const dispatch = useDispatch();
    const entityDetails = useSelector(
        ({ entityDetails }) => entityDetails && entityDetails.entity
    );

    const { hasThirdPartyRepo, patchSetName } = useSelector(
        ({ entityDetails }) => entityDetails ?? {}
    );
    const entityId = match.params?.inventoryId;
    useEffect(() => {
        dispatch(fetchSystemDetailsAction(entityId));
        return () => {
            dispatch(clearNotifications());
        };
    }, [patchSetState.shouldRefresh]);

    const pageTitle = entityDetails && `${entityDetails.display_name} - ${intl.formatMessage(messages.titlesSystems)}`;
    setPageTitle(pageTitle);

    const showBaselineModal = () => {
        setBaselineState({ isOpen: true, systemsIDs: [entityId] });
    };

    return (
        <DetailWrapper
            onLoad={({ mergeWithDetail }) => {
                register({
                    ...mergeWithDetail(SystemDetailStore)
                });
            }}
        >
            {(patchSetState.isOpen) &&
                <PatchSetWizard systemsIDs={patchSetState.systemsIDs} setBaselineState={setBaselineState} />}
            <Header
                title=""
                headerOUIA={'inventory-details'}
                breadcrumbs={[
                    {
                        title: intl.formatMessage(messages.titlesPatchSystems),
                        to: paths.systems.to,
                        isActive: false
                    },
                    entityDetails && {
                        title: entityDetails.display_name,
                        isActive: true
                    }
                ]}
            >
                <InventoryDetailHead hideBack
                    showTags
                    actions={[
                        {
                            title: intl.formatMessage(messages.titlesPatchSetAssignMultipleButton),
                            key: 'assign-to-patch-set',
                            onClick: showBaselineModal
                        }]}
                >
                    <Grid>
                        <GridItem>
                            {patchSetName && <TextContent>
                                <Text>
                                    {`${intl.formatMessage(messages.labelsColumnsPatchSet)}: ${patchSetName}`}
                                </Text>
                            </TextContent>
                            }
                        </GridItem>
                        <GridItem>
                            {hasThirdPartyRepo &&
                                (<Alert className='pf-u-mt-md' isInline variant="info"
                                    title={intl.formatMessage(messages.textThirdPartyInfo)}>
                                </Alert>)
                            }
                        </GridItem>
                    </Grid>
                </InventoryDetailHead>
            </Header>
            <Main>
                <AppInfo />
            </Main>
        </DetailWrapper>
    );
};

InventoryDetail.propTypes = {
    match: propTypes.object
};

export default InventoryDetail;
