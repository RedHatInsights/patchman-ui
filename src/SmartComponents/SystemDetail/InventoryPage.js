import React from 'react';
import { Main } from '@redhat-cloud-services/frontend-components';
import { useSelector } from 'react-redux';
import Header from '../../PresentationalComponents/Header/Header';
import { paths } from '../../Routes';
import { register } from '../../store';
import { SystemDetailStore } from '../../store/Reducers/SystemDetailStore';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';
import { setPageTitle } from '../../Utilities/Hooks';
import { InventoryDetailHead, AppInfo } from '@redhat-cloud-services/frontend-components/components/cjs/Inventory';

const InventoryDetail = () => {
    const entityDetails = useSelector(
        ({ entityDetails }) => entityDetails && entityDetails.entity
    );

    const pageTitle = entityDetails && `${entityDetails.display_name} - ${intl.formatMessage(messages.titlesSystems)}`;
    setPageTitle(pageTitle);

    return (
        <React.Fragment>
            <Header
                onLoad={({ mergeWithDetail }) => {
                    register({
                        ...mergeWithDetail(SystemDetailStore)
                    });
                }}
                title=""
                headerOUIA={'inventory-details'}
                breadcrumbs={[
                    {
                        title: intl.formatMessage(messages.titlesPatchSystems),
                        to: paths.systems.to,
                        isActive: false
                    },
                    entityDetails && {
                        title: entityDetails.display_name,
                        isActive: true
                    }
                ]}
            >
                {<InventoryDetailHead hideBack />}
            </Header>
            <Main>
                <AppInfo />
            </Main>
        </React.Fragment>
    );
};

export default InventoryDetail;
