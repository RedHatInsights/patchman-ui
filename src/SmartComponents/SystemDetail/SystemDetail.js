import { Tab, Tabs, TabTitleText } from '@patternfly/react-core';
import React from 'react';
import SystemAdvisories from '../SystemAdvisories/SystemAdvisories';
import { useSelector } from 'react-redux';
import SystemPackages from '../SystemPackages/SystemPackages';
import './SystemDetail.scss';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';
import { NoSystemData } from '../../PresentationalComponents/Snippets/NoSystemData';
import propTypes from 'prop-types';

const SystemDetail = ({ isInventoryApp }) => {
    const [activeTabKey, setActiveTabKey] = React.useState(0);
    const [areTabsHidden, setTabsHidden] = React.useState(false);

    const entity = useSelector(({ entityDetails }) => entityDetails.entity && entityDetails.entity || {});
    const onTabSelect = (event, id) => {
        setActiveTabKey(id);
    };

    const handleNoSystemData = () => {
        isInventoryApp && setTabsHidden(prevTabsHidden => !prevTabsHidden);
        return isInventoryApp && null || <NoSystemData />;
    };

    return !entity.id ? null : (!areTabsHidden && (
        <Tabs activeKey={activeTabKey} onSelect={onTabSelect} className={'patchTabSelect'} isHidden>
            <Tab eventKey={0} title={<TabTitleText>{intl.formatMessage(messages.titlesAdvisories)}</TabTitleText>}
                data-ouia-component-type={`system-advisories-tab`}
                data-ouia-component-id={`system-advisories-tab`}
            >
                <SystemAdvisories handleNoSystemData={handleNoSystemData}/>
            </Tab>
            <Tab
                eventKey={1}
                title={<TabTitleText>{intl.formatMessage(messages.titlesPackages)}</TabTitleText>}
                data-ouia-component-type={`system-packages-tab`}
                data-ouia-component-id={`system-packages-tab`}
            >
                <SystemPackages handleNoSystemData={handleNoSystemData}/>
            </Tab>
        </Tabs>
    )  || <NoSystemData/>);
};

SystemDetail.propTypes = {
    isInventoryApp: propTypes.bool
};
export default SystemDetail;
