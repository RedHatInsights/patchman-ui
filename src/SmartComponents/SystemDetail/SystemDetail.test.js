import SystemDetail from './SystemDetail';
import toJson from 'enzyme-to-json';
import { initMocks } from '../../Utilities/unitTestingUtilities.js';

/* eslint-disable */
initMocks()

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: () => ({ entity: {}})
}));

beforeEach(() => {  
   console.error = () => {};
});

describe('SystemDetail.js', () => {
    it('Should match the snapshots', () => {
        const wrapper = shallow(<SystemDetail entity = {{entity: { id: 1 }}} />); 
        expect(toJson(wrapper.update())).toMatchSnapshot();
    });

    // was not able to mock the store. 
    //Let's push this as hot fix and implement the test later

    // it('Should change the tab', () => {
    //     const wrapper = shallow(<SystemDetail />);
    //     const onSelect = wrapper.find('Tabs').props().onSelect;
    //     act(() => onSelect());
    //     expect(wrapper.find('SystemPackages')).toBeTruthy();
    // });
});
/* eslint-enable */
