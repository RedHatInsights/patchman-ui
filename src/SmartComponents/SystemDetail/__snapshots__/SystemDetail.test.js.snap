// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemDetail.js Should match the snapshots 1`] = `""`;
